website/src/parsers/html/angular.js

```diff
@@ -0,0 +1,115 @@
+import defaultParserInterface from '../utils/defaultParserInterface';
+import pkg from '@angular/compiler/package.json';
+
+const ID = 'angular';
+
+export default {
+  ...defaultParserInterface,
+
+  id: ID,
+  displayName: ID,
+  version: pkg.version,
+  homepage: pkg.homepage,
+  locationProps: new Set([
+    'span',
+    'sourceSpan',
+    'startSourceSpan',
+    'endSourceSpan',
+  ]),
+
+  loadParser(callback) {
+    require(['@angular/compiler'], callback);
+  },
+
+  parse(ng, code, options) {
+    const ast = ng.parseTemplate(code, 'astexplorer.html', options);
+    fixSpan(ast, code);
+    return ast;
+  },
+
+  nodeToRange(node) {
+    if (node.startSourceSpan) {
+      if (node.endSourceSpan) {
+        return [
+          node.startSourceSpan.start.offset,
+          node.endSourceSpan.end.offset,
+        ];
+      }
+      return [
+        node.startSourceSpan.start.offset,
+        node.startSourceSpan.end.offset,
+      ];
+    }
+    if (node.sourceSpan) {
+      return [node.sourceSpan.start.offset, node.sourceSpan.end.offset];
+    }
+    if (node.span) {
+      return [node.span.start, node.span.end];
+    }
+  },
+
+  getNodeName,
+
+  getDefaultOptions() {
+    return {
+      preserveWhitespaces: false,
+    };
+  },
+};
+
+function getNodeName(node) {
+  return node.constructor && node.constructor.name;
+}
+
+function fixSpan(ast, code) {
```

Comment:
Could you add a comment and explain what is happening here and why? For future maintainers 😉

Reply:
👍 Done. 0063ddf6d87f52dc65538945e00f10e7243e75fd